week.go

```diff
@@ -39,26 +40,96 @@ func (w *Week) Week() int {
 // Next calculates and returns the next week. If the next week is invalid (year > 9999) the function
 // returns an error.
 func (w *Week) Next() (Week, error) {
-	var year, week int
-
-	if w.week >= weeksInYear(w.year) {
-		year, week = w.year+1, 1
-	} else {
-		year, week = w.year, w.week+1
-	}
-
-	return New(year, week)
+	return w.Add(1)
 }
 
 // Previous calculates and returns the previous week. If the previous week is invalid (year < 0) the
 // function returns an error.
 func (w *Week) Previous() (Week, error) {
-	var year, week int
+	return w.Add(-1)
+}
+
+// Add calculates and returns a week that is the given positive distance (number of weeks) from the current week
+func (w *Week) Add(weeks int) (Week, error) {
+	return w.add(weeks)
+}
+
+// Sub calculates the positive difference between w and u (u-w) in number of weeks
+func (w *Week) Sub(u *Week) int {
+	var (
+		diff        = 0
```

Comment:
Using var blocks in functions can tempt to declare and initialize variables too early, which often decreases readability.